cmake/modules/AddSwift.cmake

```diff
@@ -565,8 +571,8 @@ function(_add_swift_library_single target name)
   # we add markers for metadata sections in the shared libraries using 
   # these object files.  This wouldn't be necessary if the link was done by
   # the swift binary: rdar://problem/19007002
-  if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux" OR
-     "${CMAKE_SYSTEM_NAME}" STREQUAL "FreeBSD")
+  if("${SWIFTLIB_SINGLE_SDK}" STREQUAL "Linux" OR
+      "${SWIFTLIB_SINGLE_SDK}" STREQUAL "FreeBSD")
```

Comment:
Is that the general style?  Does it make a difference here?  If not, we could uppercase them in a follow on change for style specifically.